front/pages/w/[wId]/builder/assistants/[aId]/index.tsx

```diff
@@ -160,26 +160,26 @@ export default function EditAssistant({
 }: InferGetServerSidePropsType<typeof getServerSideProps>) {
   let actionMode: AssistantBuilderInitialState["actionMode"] = "GENERIC";
 
-  let filteringMode: AssistantBuilderInitialState["filteringMode"] = null;
   let timeFrame: AssistantBuilderInitialState["timeFrame"] = null;
 
   if (isRetrievalConfiguration(agentConfiguration.action)) {
-    actionMode = "RETRIEVAL";
-    if (agentConfiguration.action?.relativeTimeFrame) {
-      switch (agentConfiguration.action.relativeTimeFrame) {
-        case "auto":
-          filteringMode = "SEARCH";
-          break;
-        case "none":
-          filteringMode = "SEARCH";
-          break;
-        default:
-          filteringMode = "TIMEFRAME";
-          timeFrame = {
-            value: agentConfiguration.action.relativeTimeFrame.duration,
-            unit: agentConfiguration.action.relativeTimeFrame.unit,
-          };
+    if (agentConfiguration.action.query === "none") {
+      if (
+        agentConfiguration.action.relativeTimeFrame === "auto" ||
+        agentConfiguration.action.relativeTimeFrame === "none"
+      ) {
+        throw new Error(
```

Comment:
Why do we throw an Error here? Shouldn't we still load the page and handle the error in `<AssistantBuilder`?

Reply:
This is a "PLS situation; this should never happen and we should be notified loudly if it does. I think it is one of the cases in which it's appropriate to throw. I can add a comment to express that though, because your question was legitimate

Reply:
How can this happen? The AssistantBuilder component should prevent it right?